akka-projection-core/src/main/scala/akka/projection/OffsetManagement.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.projection
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.concurrent.Future
+
+import akka.Done
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Extension
+import akka.actor.typed.ExtensionId
+import akka.actor.typed.pubsub.Topic
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.annotation.InternalApi
+import akka.util.Timeout
+
+object OffsetManagement extends ExtensionId[OffsetManagement] {
+  def createExtension(system: ActorSystem[_]): OffsetManagement = new OffsetManagement(system)
+
+  def get(system: ActorSystem[_]): OffsetManagement = apply(system)
+}
+
+class OffsetManagement(system: ActorSystem[_]) extends Extension {
+  private implicit val sys: ActorSystem[_] = system
+  private implicit val askTimeout: Timeout = 10.seconds // FIXME configuration
+  private implicit val ec: ExecutionContext = system.executionContext
+
+  private val topics =
+    new ConcurrentHashMap[String, ActorRef[Topic.Command[ProjectionBehavior.OffsetManagementCommand]]]()
+
+  private def topicName(projectionName: String): String =
+    "projection-" + projectionName
+
+  private def topic(projectionName: String): ActorRef[Topic.Command[ProjectionBehavior.OffsetManagementCommand]] = {
+    topics.computeIfAbsent(projectionName, _ => {
+      val name = topicName(projectionName)
+      system.systemActorOf(Topic[ProjectionBehavior.OffsetManagementCommand](name), name)
+    })
+  }
+
+  /**
+   * INTERNAL API: ProjectionBehavior registers when started.
+   */
+  @InternalApi private[akka] def register(
+      projectionId: ProjectionId,
+      projection: ActorRef[ProjectionBehavior.OffsetManagementCommand]): Unit = {
+    topic(projectionId.name) ! Topic.Subscribe(projection)
+  }
+
+  def getAllProjectionIds(projectionName: String): Future[Set[ProjectionId]] = {
+    // FIXME publish to the topic and gather all replies, or if we know that we are running with ShardedDaemonProcess we know how many
+    ???
+  }
+
+  def getOffset[Offset](projectionId: ProjectionId): Future[Option[Offset]] = {
+    topic(projectionId.name)
+      .ask[ProjectionBehavior.CurrentOffset[Offset]](replyTo =>
+        Topic.Publish(ProjectionBehavior.GetOffset(projectionId, replyTo)))
+      .map(currentOffset => currentOffset.offset)
+  }
```

Comment:
we could also add a convenience for retrieving all offsets as suggested in the issue:

> Method to retrieve all offsets for all processors in database (nice to have, could be done by retrieving one at a time, but for a projection status management call, typically all will just be retrieved).

However, I wonder if tracking of current offset shouldn't be done by the status tracking by https://github.com/akka/akka-projection/pull/185 since an Admin UI is probably interested in the error tracking.

We could also integrate the status tracking with this management API, so that the current status can also be retrieved via this management API.